api/src/main/java/org/ccci/idm/user/User.java

```diff
@@ -241,6 +241,20 @@ public void setFirstName(final String firstName) {
         this.firstName = firstName;
     }
 
+    @Nullable
+    public String getRawPreferredName() {
+        return preferredName;
+    }
+
+    @Nullable
+    public String getPreferredName() {
+        return !Strings.isNullOrEmpty(preferredName) ? preferredName : firstName;
+    }
+
+    public void setPreferredName(@Nullable final String name) {
+        preferredName = Objects.equal(firstName, name) ? null : name;
```

Comment:
Without this functionality preferredName will effectively not have an unset state because we store the fully resolved preferredName within eDirectory. So, once we save the user once and reload the user, the preferredName will be set no matter what.